File: Curso-ciencias-de-la-computacion/12.-Sorting/Selection sort/Python/main.py
#Selection sort implemented in python by JCPC
#Time complexity: O(n^2)

def selectionSort(array):
    for i in range(0, len(array)):
        min = i
        for j in range(i, len(array)):
            if array[j] < array[min]:
                min = j
        array[i], array[min] = array[min], array[i]

def main():
    array = [3, 5, 1, 9, 6, 17, 7, 89, 0]
    array = selectionSort(array)
    print(array)

if __name__=="__main__":
    main()
            

File: Curso-ciencias-de-la-computacion/12.-Sorting/Selection sort/C/main.c
//Selection sort algorithm in C
//By JCPC

#include <stdio.h>
#include <stdbool.h>

#define ARR_LEN 7

void selectionSort(int arr[]){
    int min = 0;
    int temp = 0;
    for(int i = 0; i < ARR_LEN; i++){
        min = i;
        for(int j = i; j < ARR_LEN; j++){
            if(arr[j] < arr[min]){
                min = j;
            }
        } 
        temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
    }   
}

int main(){
    int arr[] = { 9, 4, 3, 8, 10, 2, 5 };
    selectionSort(arr);

    for(int i = 0; i < ARR_LEN; i++){
        printf("%d ", arr[i]);
    }
    return 0;
}

